src/reducers/visualizations.js

```diff
@@ -1,29 +1,48 @@
+import { orObject } from '../util';
+import objectClean from 'd2-utilizr/lib/objectClean';
+
 /** @module reducers/visualizations */
 
+const isEmpty = p => p === undefined || p === null;
+
 export const actionTypes = {
     RECEIVED_VISUALIZATION: 'RECEIVED_VISUALIZATION',
+    RECEIVED_ACTIVE_VISUALIZATION: 'RECEIVED_ACTIVE_VISUALIZATION',
 };
 
 export default (state = {}, action) => {
     switch (action.type) {
         case actionTypes.RECEIVED_VISUALIZATION: {
-            const res = Object.assign({}, state, {
+            return {
+                ...state,
                 [action.value.id]: action.value,
-            });
-
-            return res;
+            };
+        }
+        case actionTypes.RECEIVED_ACTIVE_VISUALIZATION: {
```

Comment:
Please add a reducer unit test for the new action

Reply:
Done